test-framework/test-suites/integration/tests/report/test_report_host_bootfile.py

```diff
@@ -0,0 +1,145 @@
+import pytest
+
+@pytest.mark.usefixtures('revert_database')
+class TestReportHostBootfile:
+	def test_report_host_bootfile_on_fresh_install(self, host):
+		result = host.run('stack report host bootfile')
+		assert result.rc == 0
+
+	def test_report_host_bootfile_for_host_with_full_IP_config(self, host):
+		result = host.run('stack add host backend-1-1')
+		assert result.rc == 0
+		result = host.run('stack add network public address=192.172.0.0 mask=255.255.255.0 pxe=True')
+		assert result.rc == 0
+		result = host.run('stack add host interface backend-1-1 interface=eth1 ip=192.172.0.2 network=public default=True mac=52:54:00:6e:50:91')
+		assert result.rc == 0
+
+		result = host.run('stack report host bootfile')
+		assert result.rc == 0
+		result_specific = host.run('stack report host bootfile backend-1-1')
+		assert result_specific.rc == 0
+		assert result.stdout == result_specific.stdout
```

Comment:
You're checking the outputs match, which is great, but not that there actually *was or was not* output.

Likewise, while we're writing tests for this command, we should validate that the command produces different output for `action=os` vs `action=install`.